fauna/v10/queries/deploy.js

```diff
@@ -1,44 +1,68 @@
-const { fql, Module } = require("fauna");
+const { fql } = require("fauna");
 
 /**
- * Constructs an FQL Query to create/update resources according to the passed parameters.
+ * Constructs an FQL Query to create/update functions according to the passed parameters.
  *
- * @param module The Module to create or update. E.g. Module("Collection")
  * @param params The params to pass to the create or update call.
- * @returns An FQL X Query. The query returns an object:
- *          { type: "Function", name: str, result: "created" | "updated" | "noop" }
+ * @param preview A boolean flag to indicate whether to mutate or just log.
+ * @returns An FQL Query. The query returns an array of results:
+ *          [{ type: "Function", name: str, action: "created" | "updated", preview: bool}, ...]
  */
-const createUpdate = (module, params) => {
+const createUpdateFunction = (params, preview = false) => {
   return fql`
   {
     let p = ${params}
-    let mod = ${module}
-    if (mod.byName(p.name) != null) {
-      let f = mod.byName(p.name)
-      if (Object.entries(p).every(v => v[1] == f[v[0]])) {
-        { type: mod.toString(), name: p.name, result: "noop" }
-      } else {
-        if (p.data != f.data) {
-          let pNullData = Object.assign(p, { data: null })
-          f.update(pNullData)
+    let p = p { name, body, data, role, }
+    
+    if (Function.byName(p.name) != null) {
```

Comment:
style: putting a != and an else together makes a double negative in my head, so I'd switch around the ordering of the if/else. super minor though, this is just my preference

Reply:
nice, i like that. update coming